src/ralph/assets/models/networks.py

```diff
@@ -0,0 +1,389 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import ipaddr
+
+from django.db import models
+from django.core.exceptions import ValidationError
+from django.utils.encoding import python_2_unicode_compatible
+from django.utils.translation import ugettext_lazy as _
+
+from ralph.assets.models.assets import Asset
+from ralph.assets.models.mixins import (
+    LastSeenMixin,
+    NamedMixin,
+    TimeStampMixin,
+)
+
+
+def network_validator(value):
+    try:
+        ipaddr.IPv4Network(value)
+    except ipaddr.NetmaskValueError as exc:
+        raise ValidationError(exc.message)
+
+
+@python_2_unicode_compatible
+class NetworkKind(NamedMixin, models.Model):
+    class Meta:
+        verbose_name = _('network kind')
+        verbose_name_plural = _('network kinds')
+        ordering = ('name',)
+
+    def __str__(self):
+        return self.name
+
+
+@python_2_unicode_compatible
+class NetworkEnvironment(NamedMixin):
+    # TODO: more generic
+    # data_center = models.ForeignKey(
+    #     'DataCenter',
+    #     verbose_name=_('data center')
+    # )
+    queue = models.ForeignKey(
+        'DiscoveryQueue',
+        verbose_name=_('discovery queue'),
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+    )
+    hosts_naming_template = models.CharField(
+        max_length=30,
+        help_text=_(
+            'E.g. h<200,299>.dc|h<400,499>.dc will produce: h200.dc '
+            'h201.dc ... h299.dc h400.dc h401.dc'
+        ),
+    )
+    next_server = models.CharField(
+        max_length=32,
+        blank=True,
+        default='',
+        help_text=_('The address for a TFTP server for DHCP.'),
+    )
+    domain = models.CharField(
+        _('domain'),
+        max_length=255,
+        blank=True,
+        null=True,
+    )
+    remarks = models.TextField(
+        verbose_name=_('remarks'),
+        help_text=_('Additional information.'),
+        blank=True,
+        null=True,
+    )
+
+    def __str__(self):
+        return self.name
+
+    class Meta:
+        ordering = ('name',)
+
+
+@python_2_unicode_compatible
+class AbstractNetwork(models.Model):
```

Comment:
why this is abstract? couldn't it be just one simple not-abstract `Network` model (since there is only one model which is inheriting `AbstractNetwork` and it's `Network`).